### PR TITLE
Integrated react-native-size-matters library

### PR DESCRIPTION
**Pull request by sangameshsomawar**
Fixes: #537 
https://github.com/bigbinary/neeto-ui-rn/issues/489

### CODE REVIEW

**Comment by sangameshsomawar** [comment]
@kamolesh97 @chiragbhaiji  I have scaled almost all components, Some components are not looking good after scaling. 
I will fix those and will test via yalc in other apps once.

**Comment by sangameshsomawar** [comment]
Bugs:

1. Fix the tab bar size issue in iPhone SE, Ref: https://github.com/bigbinary/neeto-chat-rn/pull/514#pullrequestreview-1206452841

**Comment by sangameshsomawar** [comment]
@chiragbhaiji @kamolesh97 Please review

File: src/components/BottomSheet.jsx [hunk 1]
```diff
@@ -67,7 +72,7 @@ const Title = ({
       </Container>
       {canSearch && (
         <SearchBar
-          debounceDelay={200}
+          debounceDelay={moderateScale(200)}
```
Comment (chiragbhaiji) [inline src/components/BottomSheet.jsx:75]:
```suggestion
          debounceDelay={200}
```

File: src/components/Card.jsx [hunk 2]
```diff
@@ -80,10 +81,10 @@ export const styles = StyleSheet.create({
...
       width: 0,
-      height: 2,
+      height: moderateScale(2),
     },
-    shadowOpacity: 0.25,
-    shadowRadius: 3.84,
-    elevation: 5,
+    shadowOpacity: moderateScale(0.25),
+    shadowRadius: moderateScale(3.84),
+    elevation: moderateScale(5),
```
Comment (chiragbhaiji) [inline src/components/Card.jsx:88]:
```suggestion
    elevation: 5,
```

File: src/components/Card.jsx [hunk 3]
```diff
@@ -80,10 +81,10 @@ export const styles = StyleSheet.create({
     shadowColor: theme.colors.background.grey800,
     shadowOffset: {
       width: 0,
-      height: 2,
+      height: moderateScale(2),
     },
-    shadowOpacity: 0.25,
-    shadowRadius: 3.84,
-    elevation: 5,
+    shadowOpacity: moderateScale(0.25),
```
Comment (chiragbhaiji) [inline src/components/Card.jsx:86]:
```suggestion
    shadowOpacity: 0.25,
```

File: src/components/Card.jsx [hunk 4]
```diff
@@ -80,10 +81,10 @@ export const styles = StyleSheet.create({
...
+    shadowRadius: moderateScale(3.84),
```
Comment (chiragbhaiji) [inline src/components/Card.jsx:87]: @sangameshsomawar should we also revert it? I think yes.

File: src/components/FAB.jsx [hunk 5]
```diff
@@ -50,16 +51,16 @@ export const FAB = ({ Icon, bg, disabled, variant, onPress, ...rest }) => {
...
       width: 0,
-      height: 2,
+      height: moderateScale(2),
     },
-    shadowOpacity: 0.25,
-    shadowRadius: 3.84,
-    elevation: 5,
+    shadowOpacity: moderateScale(0.25),
+    shadowRadius: moderateScale(3.84),
+    elevation: moderateScale(5),
```
Comment (chiragbhaiji) [inline src/components/FAB.jsx:58]:
```suggestion
    elevation: 5,
```

File: src/components/FAB.jsx [hunk 6]
```diff
@@ -50,16 +51,16 @@ export const FAB = ({ Icon, bg, disabled, variant, onPress, ...rest }) => {
     shadowColor: theme.colors.background.grey800,
     shadowOffset: {
       width: 0,
-      height: 2,
+      height: moderateScale(2),
     },
-    shadowOpacity: 0.25,
-    shadowRadius: 3.84,
-    elevation: 5,
+    shadowOpacity: moderateScale(0.25),
```
Comment (chiragbhaiji) [inline src/components/FAB.jsx:56]:
```suggestion
    shadowOpacity: 0.25,
```

File: src/components/FAB.jsx [hunk 7]
```diff
@@ -50,16 +51,16 @@ export const FAB = ({ Icon, bg, disabled, variant, onPress, ...rest }) => {
...
+    shadowRadius: moderateScale(3.84),
+    elevation: moderateScale(5),
   };
 
   let style = rest.style || {};
   if (disabled) {
-    style = { ...style, opacity: 0.5 };
+    style = { ...style, opacity: moderateScale(0.5) };
```
Comment (chiragbhaiji) [inline src/components/FAB.jsx:63]:
```suggestion
    style = { ...style, opacity: 0.5 };
```

File: src/components/Input.jsx [hunk 8]
```diff
@@ -160,18 +154,19 @@ export const Input = ({
           : isFocused
           ? colors.border.purple500
           : colors.border.grey400,
-        borderWidth: errorMessage || isFocused ? 1.5 : 1,
+        borderWidth: errorMessage || isFocused ? 1.5 : moderateScale(1),
```
Comment (chiragbhaiji) [inline src/components/Input.jsx:157]:
```suggestion
        borderWidth: errorMessage || isFocused ? moderateScale(1.5) : moderateScale(1),
```

**Review by chiragbhaiji — APPROVED** [review]
All looks good now.

**Review by kamolesh97 — APPROVED** [review]
@sangameshsomawar LGTM as well.

**Pull request merged**